File: src/main/java/com/github/thaynarasilvapinto/SimuladorBanco/repositories/OperacaoRepository.java
package com.github.thaynarasilvapinto.SimuladorBanco.repositories;

import com.github.thaynarasilvapinto.SimuladorBanco.domain.Conta;
import com.github.thaynarasilvapinto.SimuladorBanco.domain.Operacao;
import com.github.thaynarasilvapinto.SimuladorBanco.domain.TipoOperacao;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import javax.swing.*;
import java.util.List;

@Repository
public interface OperacaoRepository extends JpaRepository<Operacao, Integer> {

    List<Operacao> findAllByContaOrigem(Conta conta);
    List<Operacao> findAllByContaDestinoAndTipoOperacao(Conta conta, TipoOperacao operacao);
    List<Operacao> findAllByContaOrigemAndTipoOperacao(Conta conta, TipoOperacao operacao);

}

File: src/main/kotlin/com/github/thaynarasilvapinto/simuladorbanco/repositories/OperacaoRepository.kt
package com.github.thaynarasilvapinto.simuladorbanco.repositories

import com.github.thaynarasilvapinto.simuladorbanco.domain.Conta
import com.github.thaynarasilvapinto.simuladorbanco.domain.Operacao
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface OperacaoRepository : JpaRepository<Operacao, Int> {

    fun findAllByContaOrigem(conta: Conta): List<Operacao>
    fun findAllByContaDestinoAndTipoOperacao(conta: Conta, operacao: TipoOperacao): List<Operacao>
    fun findAllByContaOrigemAndTipoOperacao(conta: Conta, operacao: TipoOperacao): List<Operacao>

}